src/App.tsx

```diff
@@ -1,26 +1,39 @@
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
+import { useEffect } from 'react';
+import { getTodos } from './api';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import { todosSlice } from './features/todos';
 import { Loader, TodoFilter, TodoList, TodoModal } from './components';
 
-export const App = () => (
-  <>
-    <div className="section">
-      <div className="container">
-        <div className="box">
-          <h1 className="title">Todos:</h1>
+export const App = () => {
+  const hasTodos = useAppSelector(state => state.todos.length > 0);
+  const hasSelectedTodo = useAppSelector(state => state.currentTodo !== null);
+  const dispatch = useAppDispatch();
 
-          <div className="block">
-            <TodoFilter />
-          </div>
+  useEffect(() => {
+    getTodos().then(todosFromServer =>
+      dispatch(todosSlice.actions.set(todosFromServer)),
+    );
+  }, [dispatch]);
```

Comment:
The `useEffect` dependency array includes `dispatch`, which is technically correct but unnecessary since `dispatch` is a stable reference from `useAppDispatch`. You can safely omit it from the dependency array.